tasks/build/preconfigure-k3s.yml

```diff
@@ -113,13 +122,6 @@
   when: k3s_registration_address is not defined
         or k3s_control_delegate is not defined
 
-- name: Ensure k3s_runtime_config is set for control plane
-  ansible.builtin.set_fact:
-    k3s_runtime_config: "{{ (k3s_server | default({})) | combine((k3s_agent | default({}))) }}"
-  when:
-    - (k3s_server is defined or k3s_agent is defined)
-    - (k3s_control_node is defined and k3s_control_node)
-
 - name: Ensure k3s_runtime_config is set for agents
```

Comment:
I've pushed a slight tweak, [this var, `k3s_runtime_config`](https://github.com/PyratLabs/ansible-role-k3s/blob/main/vars/main.yml#L59) combines `k3s_agent` and `k3s_server` on all node - so this can be used, de-duplicating the task of setting from `node-ip`. In addition, there was a `set_fact` that duplicates the setting of the var.

Looks good to me :+1: